lib/kubeclient/common.rb

```diff
@@ -425,6 +425,7 @@ def api
     def load_entities
       @entities = {}
       fetch_entities['resources'].each do |resource|
+        resource = Kubeclient::Common::Compatibility31.compatible_resource(resource)
```

Comment:
Yeah I'm good with that (.nil? since empty comes from rails) I'd definitely like to see a backward compatibility test.
@zakiva take a look at test/json/core_api_resource_list.json and it's use all over. You would probably want to rename it to core_api_resource_list_v13.json and add a core_api_resource_list_v11.json that you can test in a new file.